Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/DynamicNode/DynamicNodeSlotConfig.h

```diff
@@ -34,11 +35,19 @@ namespace AtomToolsFramework
             const AZStd::string& displayName,
             const AZStd::string& description,
             const AZStd::any& defaultValue,
-            const AZStd::vector<AZStd::string>& supportedDataTypes,
+            const AZStd::string& supportedDataTypeRegex,
             const DynamicNodeSettingsMap& settings);
         DynamicNodeSlotConfig() = default;
         ~DynamicNodeSlotConfig() = default;
 
+        AZ::Crc32 ValidateDataTypes();
+        AZStd::any GetDefaultValue() const;
+        AZStd::string GetDefaultDataTypeName() const;
+        GraphModel::DataTypePtr GetDefaultDataType() const;
+        AZStd::vector<AZStd::string> GetSupportedDataTypeNames() const;
```

Comment:
I'd like a comment about how this relates to supportedDataTypeRegex that's passed to the constructor.